content/blog/part-8-agentic-ai-and-qdrant-building-semantic-memory-with-mcp-protocol.md

```diff
@@ -0,0 +1,218 @@
+    |-- [MCP Client]
+[MCP Protocol]
+    |
+    |-- [Qdrant MCP Server]
+    |   |-- Tool: qdrant-store
+    |   |-- Tool: qdrant-find
+    |
+[Qdrant Vector DB]
+```
+
+### Use Case: Support Ticket Memory for AI Assistants
+
+Imagine an AI assistant answering support queries.
+
+* It doesn't have all answers built-in.
+* But it has semantic memory from prior support logs stored in Qdrant.
+* It uses qdrant-find to semantically retrieve similar issues and then formulates a contextual response.
+
+## Step-by-Step Implementation
+
+### Step 1: Launch Qdrant MCP Server
+
+```cwl
+export COLLECTION_NAME="support-tickets"
+export QDRANT_LOCAL_PATH="./qdrant_local_db"
+export EMBEDDING_MODEL="sentence-transformers/all-MiniLM-L6-v2"
+```
+
+```cwl
+uvx mcp-server-qdrant --transport sse
+```
+
+## Key Parameters:
+
+* COLLECTION_NAME: Name of the Qdrant collection
+* QDRANT_LOCAL_PATH: Local vector DB storage path
+* EMBEDDING_MODEL: Embedding model for vectorization
+
+### Step 2: Connect the MCP Client
+
+```python
+from mcp import ClientSession, StdioServerParameters
+from mcp.client.stdio import stdio_client
+async def main():
+server_params = StdioServerParameters(
+     command="uvx",
+     args=\["mcp-server-qdrant"],
+     env={
+         "QDRANT_LOCAL_PATH": "./qdrant_local_db",
+         "COLLECTION_NAME": "support-tickets",
+         "EMBEDDING_MODEL": "sentence-transformers/all-MiniLM-L6-v2"
+     }
+)
+```
+
+```python
+async with stdio_client(server_params) as (read, write):
+     async with ClientSession(read, write) as session:
+         await session.initialize()
+         tools = await session.list_tools()
+         print(tools)
+```
+
+```cwl
+Expected Output: Lists tools like qdrant-store, qdrant-find
+```
+
+### Step 3: Ingest a New Memory
+
+```python
+ticket_info = "Order #1234 was delayed due to heavy rainfall in transit zone."
+result = await session.call_tool("qdrant-store", arguments={
+"information": ticket_info,
+"metadata": {"order_id": 1234}
+})
+```
+
+This stores an embedded version of the text in Qdrant.
+
+### Step 4: Perform a Semantic Search
+
+```python
+query = "Why was order 1234 delayed?"
+search_response = await session.call_tool("qdrant-find", arguments={
+"query": "order 1234 delay"
+})
+```
+
+## Example Output:
+
+```cwl
+[
+  {
+"content": "Order #1234 was delayed due to heavy rainfall in transit zone.",
+"metadata": {"order_id": 1234}
+  }
+]
+```
+
+### Step 5: Use with LLM
+
+```python
+import openai
+context = "\n".join(\[r["content"] for r in search_response])
+prompt = f"""
+You are a helpful assistant. Use this context to answer:
+"""
+{context}
+"""
+Question: Why was order #1234 delayed?
+"""
+response = openai.ChatCompletion.create(
+model="gpt-3.5-turbo",
+messages=[{"role": "user", "content": prompt}]
+)
+print(response\["choices"]\[0]\["message"]\["content"])
+```
+
+### Final Answer:
+
+```cwl
+"Order #1234 was delayed due to heavy rainfall in the transit zone."
+```
+
+## Parameter Reference
```

Comment:
Change to "Parameter reference"

Also, just check to see if this is formatted the way you want. It might be, but I'm not sure:

<img width="1490" height="668" alt="image" src="https://github.com/user-attachments/assets/f39030f4-7292-4aee-89c4-7abd46748d29" />